platform/darwin/src/NSData+MGLAdditions.mm

```diff
@@ -0,0 +1,33 @@
+#import "NSData+MGLAdditions.h"
+
+#include <mbgl/util/compression.hpp>
+
+@implementation NSData (MGLAdditions)
+
+- (NSData *)mgl_compressedData
+{
+    NSString *dataString = [[NSString alloc] initWithData:self encoding:NSUTF8StringEncoding];
```

Comment:
You can avoid this step by passing `self.bytes` into a `std::string` constructor.